internal/codegen/cel_types.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2021 Zenauth Ltd.
```

Comment:
What do you think about putting this in a separate package `internal/codegen/types`? E.g. `internal/codegen/types/hierarchy.go`

Reply:
Good idea.

Reply:
I got carried away and moved almost all of CEL code (those not directly related to code generation) out of the `codegen` package to a `conditions` package. Then I realised that I would break your pending PR with that 🤦🏽 

https://github.com/cerbos/cerbos/compare/main...charithe:enhancement/cel_hierarchy_funcs_refactor

I think we should probably defer that refactoring until both PRs are merged. WDYT?

Reply:
Yes, probably there will be fewer merge conflicts if we merge my PR before that refactoring.